### PR TITLE
[SCB-245] add dubbo support for omega, pass omega context from consumer…

### PR DESCRIPTION
**Pull request by wuzunqian**
… to provider

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboProviderFilter.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
+package org.apache.servicecomb.saga.omega.transport.dubbo;
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboProviderFilter.java]: Please add the License header here.

File: omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboProviderFilter.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import org.slf4j.LoggerFactory;
+
+import java.lang.invoke.MethodHandles;
+
+import static org.apache.servicecomb.saga.omega.context.OmegaContext.GLOBAL_TX_ID_KEY;
+import static org.apache.servicecomb.saga.omega.context.OmegaContext.LOCAL_TX_ID_KEY;
+
+/**
+ * get saga transaction id from dubbo invocation and set into omega context
+* @author wuzq
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboProviderFilter.java]: We normally don't put the author and email into our code, as we can trace the code from git.

File: omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboConsumerFilter.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+import org.slf4j.LoggerFactory;
+
+/**
+ * add saga transaction id to dubbo invocation
+* @author wuzq
+* @email wuzunqian@msn.com
+* @date 03/05/2018 10:44 AM
+*/
+@Activate(
+        group = {"consumer"}
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboConsumerFilter.java]: You may use the constant here.

File: omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboConsumerFilter.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+//
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-dubbo/src/main/java/org/apache/servicecomb/saga/omega/transport/dubbo/SagaDubboConsumerFilter.java]: Please add the ASF License Header here.

File: omega/omega-transport/omega-transport-dubbo/src/main/resources/META-INF/dubbo/com.alibaba.dubbo.rpc.Filter [hunk 5]
```diff
@@ -0,0 +1,2 @@
+sagaDubboConsumerFilter=org.apache.servicecomb.saga.omega.transport.dubbo.SagaDubboConsumerFilter
```
Comment (WillemJiang) [inline omega/omega-transport/omega-transport-dubbo/src/main/resources/META-INF/dubbo/com.alibaba.dubbo.rpc.Filter]: We need to add the License header here.

**Comment by wuzunqian** [comment]
@WillemJiang I have a demo for myself, I think it's not simple enough to submit. By the way, I found a bug, but I don't know if it happen only with dubbo, I will read code and try to fix it, it may take some time.

**Comment by WillemJiang** [comment]
@wuzunqian  Please ping me at the [gitter](https://gitter.im/ServiceCombUsers/Lobby). Maybe we can find out a quick fix there.

**Pull request merged**